staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -187,6 +188,9 @@ type LeaderElector struct {
 	// clock is wrapper around time to allow for less flaky testing
 	clock clock.Clock
 
+	// used to lock the observedRecord
+	observedRecordLock sync.RWMutex
```

Comment:
sync.Mutex would be better if you don't expect the record to be read frequently.

Reply:
Thank you 🙏 : https://github.com/kubernetes/kubernetes/pull/97419/commits/1f43ab4d926644445c3c66a0fe72e688ee0b892e